### ISSUE TITLE
'vela dry-run' error when installing kubevela to non-vela-system namespace

### ISSUE THREAD

**Issue by nuclearwu** [kubevela/kubevela#4921]
**Describe the bug**

'vela dry-run -f guestbook-app.yaml' error when installing kubevela to non-vela-system namespace, the error message is workloaddefinitions.core.oam.dev “k8s-objects” not found. but specifying namespace like this 'vela dry-run -f guestbook-app.yaml -n karmada-system' works well. So I wonder if this is a problem, In theory, no matter which namespace kubevela is installed in, it should know and find objects such as workloaddefinitions.core.oam.dev "k8s-objects" when executing vela dry-run.
![image](https://user-images.githubusercontent.com/37976659/197956939-5594e4dc-81d6-4289-b16e-4bb5f8922b0c.png)
![image](https://user-images.githubusercontent.com/37976659/197957799-a1102280-a2e2-442c-9740-f3992b4829d2.png)

**To Reproduce**

**Expected behavior**

**Screenshots**

**KubeVela Version**

**Cluster information**

**Additional context**

**Comment by wonderflow** [comment]
Yes, currently the CLI will use `vela-system` as the system capability namespace, we should support to override it in some mechanism. In your case, we should support somthing such like:

```
export DefaultKubeVelaNS="karmada-system" vela dry-run -f guestbook-app.yaml
```

**Comment by nuclearwu** [comment]
> Yes, currently the CLI will use `vela-system` as the system capability namespace, we should support to override it in some mechanism. In your case, we should support somthing such like:
> 
> ```
> export DefaultKubeVelaNS="karmada-system" vela dry-run -f guestbook-app.yaml
> ```

I was wondering if it is possible to import the environment variables of the default installation namespace when specifying a non-a namespace to install vela-core, and then read the environment variables directly during vela dry-run, if the environment variables exist, use the environment variables as The default namespace, if it does not exist, the default vela-system namespace is used

**Comment by wonderflow** [comment]
yes, it's also one of the solutions. Vela-core already supports different namespace other than `vela-system`.

**Comment by pbxie** [comment]
After executing the command export DefaultKubeVelaNS="my-vela-system" vela addon list, there was no output.
![image](https://github.com/kubevela/kubevela/assets/37398509/ee5f2712-c03f-4d80-a9cc-b6c3a1d51176)

**Comment by neteric** [comment]
> After executing the command export DefaultKubeVelaNS="my-vela-system" vela addon list, there was no output. ![image](https://user-images.githubusercontent.com/37398509/256114343-ee5f2712-c03f-4d80-a9cc-b6c3a1d51176.png)

+1

**Comment by TheGthr** [comment]
Instead of a variable, would it be more consistent with the `vela install` command to have a `-n` and/or `--namespace` flag for all the other commands?

**Comment by Jamel-jun** [comment]
卧槽，我安装到其他命名空间，直接把我命名空间删了。
vela uninstall -n xx
